### PR TITLE
Java 11 compliance

### PR DESCRIPTION
**Pull request by dev-gaur**
#1478

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -42,6 +42,20 @@ jobs:
         paths:
         - ~/.m2
 
+  build-java11:
```
Comment (rohanKanojia) [inline .circleci/config.yml]: You need to add these jobs in workflows.all.jobs array also

File: .circleci/config.yml [hunk 2]
```diff
@@ -59,6 +73,23 @@ jobs:
         paths:
         - ~/.m2
 
+  # Run Tests
+  tests-java11:
+    working_directory: ~/fabric8io/fabric8-maven-plugin
+    machine: true
```
Comment (rohanKanojia) [inline .circleci/config.yml]: How are we ensuring that we're building on Java 11? I don't see any jdk11 image being used.

Comment (dev-gaur) [reply .circleci/config.yml]: The -Pjava11 flag, this profile has configured maven-compiler-plugin to use java 11.

File: parent/pom.xml [hunk 3]
```diff
@@ -934,6 +938,53 @@
       </dependencies>
     </profile>
 
+    <profile>
```
Comment (rohanKanojia) [inline parent/pom.xml]: What difference does this profile make? I compiled project on jdk11; apart from `ThorntailV2HealthCheckEnricherTest.java` project build fine. Perhaps I missed something?

Comment (lordofthejars) [reply parent/pom.xml]: J11 removes some API from J10 (ie XML) so you need to add them manually. I cannot see the file but I guess this is what you mean in your comment.

**Pull request merged**